### PR TITLE
Glossary terms in written content 

### LINKED ISSUE #110: add glossary definitions
## Use Case

[the design](https://www.figma.com/design/KEEHjYRxL9ouTuceTtyC7G/2024-Feb-Interactive-Publication?node-id=1144-121956&t=D89nZasE04jC9eew-0) has certain terms included in a glossary, and when they appear for the first time, they are rendered as link. 

## Acceptance Criteria
the glossary terms should be rendered as a link, on click show a popover that displays the definition, which closes on click, or on other interactions. 

## Details

_Please provide any helpful specifications_

### PR DESCRIPTION
**Pull request by meganrm**
Problem
=======
Estimated review size: medium
closes #110 

Solution
========
I needed to have the content render as jsx instead of just being a string. I added the glossary terms as one large object (we also have a design to show them all as a single list). 

Bonus changes:
1. This app involves a lot of z stacking, so I parametrized that a little bit so it's clearer what is in what layer. A number of the files changed are only involved in update the z-index. I suggest looking through those files first, marking them as read, and then going through the changes for the glossary term definitions 
2. Rendering the content as JSX allows for rendering the `A` `B` as they're colored icons instead of just text. 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Screenshots (optional):
-----------------------
Screenshot 2024-07-31 at 3 31 00 PM

### CODE REVIEW

File: src/content/glossary.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+export const terms = {
+    affinity: {
+        definition:
+            "Measure of the strength of an interaction between two molecules. Generally, the more chemical bonds that form between two molecules, the stronger the affinity of those molecules for each other.",
+        link: "[Learn more]",
```
Comment (tyler-foster) [inline src/content/glossary.tsx:5]: What's the difference between `[Learn more]` and `undefined` in this context? Are these ones supposed to be links later?

File: src/content/glossary.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    KD: {
+        definition:
+            "The equilibrium constant. The balance of products and reactants at equilibrium, defined mathematically as the ratio of the dissociation and association rate constants, which quantitatively measures affinity.",
+        link: "https://www.khanacademy.org/science/ap-chemistry-beta/x2eef969c74e0d802:equilibrium/x2eef969c74e0d802:magnitude-and-properties-of-the-equilibrium-constant/v/magnitude-of-the-equilibrium-constant",
+    },
+    equilibrium: {
+        definition:
+            "The state in which the amount of reactants and products of a reaction is not changing over time.",
```
Comment (tyler-foster) [inline src/content/glossary.tsx:14]: Nit: I read this weirdly the first time. Should this be rephrased? I wasn't sure if it was:
"**amount** of reactants and products of a reaction **is**"
or
"_amount of reactants_ **and** _products of a reaction_ **are"**

File: src/components/icons/LinkOut.tsx
Comment (tyler-foster) [inline src/components/icons/LinkOut.tsx:1]: Where / how does this component get used? I don't see a place where it's imported.

File: src/content/glossary.tsx (hunk 1 above)
Comment (meganrm) [reply src/content/glossary.tsx]: those originally had links in the design/copy but after I checked, we're not longer have external links for them so I removed it

File: src/components/icons/LinkOut.tsx [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
 
+/***
+ * NOTE: this currently gives a warning in the console:
+ * "Warning: FontAwesomeIcon: Support for defaultProps will be removed from function components
+ * in a future major release. Use JavaScript default parameters instead."
+ * I didn't write an issue because I can't reproduce it in codepen:
+ * https://codepen.io/meganrm/pen/BagJvXZ
+ */
+const LinkOut = () => (
+    <FontAwesomeIcon style={{ marginLeft: 6 }} icon={faUpRightFromSquare} />
```
Comment (ShrimpCryptid) [inline src/components/icons/LinkOut.tsx:12]: Consider adding some visually hidden text that says "(opens in new tab)" as part of this icon? Or add alt text to it that describes that.

https://www.w3.org/TR/WCAG20-TECHS/H83.html
https://webaim.org/techniques/css/invisiblecontent/

**Review by ShrimpCryptid — APPROVED** [review]
Looks good! Thank you for making those changes!

**Review by tyler-foster — APPROVED** [review]

**Pull request merged**